specs/Floating-Point-Extra.md

```diff
@@ -0,0 +1,108 @@
+- UNDERFLOW: If an operation produces a value that is too small for the destination format
+- INEXACT: If an operation with a rational result produces a value that cannot be exactly represented in the destiniation format.
+- INEXACT: If the operation is performed with greater precision than the destination format, and the result cannot be exactly represented in the destiniation format
+- UNDERFLOW: If the result would be a subnormal value and fpcrw.DENORM is clear.
+
+
+Flags: If `f` is not set in `h`, `N`, `Z`, and `P` are set according to the result. 
+
+Instructions:
+- 0x100 (exp): Computes exp(x) of the operand, and stores the result in the operand.
+- 0x101 (ln): Computes ln(x) (the natural logarithm) of the operand, and stores the result in the operand
+- 0x102 (lg): Computes lg(x) (the base 2 logarithm) of the operand, and stores the result in the operand.
+- 0x103 (sin): Computes sin(x) of the operand, and stores the result in the operand.
+- 0x104 (cos): Computes cos(x) of the operand, and stores the result in the operand
+- 0x105 (tan): Computes tan(x) of the operand, and stores the result in the operand.
+- 0x106 (asin): Computes asin(x) of the operand, and stores the result in the operand.
+- 0x107 (acos): Computes acos(x) of the operand, and stores the result in the operand.
+- 0x108 (atan): Computes atan(x) of the operand, and stores the result in the operand.
+- 0x109 (sinh): Computes sinh(x) (the hyperbolic sin) of the operand, and stores the result in the operand.
+- 0x10a (cosh): Computes cosh(x) (the hyperbolic cos) of the operand, and stores the result in the operand.
+- 0x10b (tanh): Computes tanh(x) (the hyperbolic tan) of the operand, and stores the result in the operand.
+- 0x10c (asinh): Computes asinh(x) (the inverse hyperbolic sin) of the operand, and stores the result in the operand
+- 0x10d (acosh): Computes acosh(x) (the inverse hyperbolic cos) of the operand, and stores the result in the operand.
+- 0x10e (atanh): Computes atanh(x) (the inverse hyperbolic tan) of the operand, and stores the result in the operand.
+- 0x10f (exp2): Computes 2^x of the operand, and stores the result in the operand.
+- 0x113 (log10): Computes log10(x) (the base 10 logarithm) of the operand, and stores the result in the operand
+- 0x114 (lnp1): Computes ln(x+1) of the operand, and stores the result in the operand
+- 0x115 (expm1): Computes exp(x-1) of the operand, and stores the result in the operand.
+- 0x119 (sqrt): Computes sqrt(x) of the operand, and stores the result in the operand.
+- 0x11a (cbrt): Computes cbrt(x) of the operand, and stores the result in the operand.
+- 0x125 (hypot): Computes the hypotenuse of the two operands and stores the result in the first.
+- 0x126 (atan2): Computes the arc tangent of the first operand, divided by the second, but use the sign of both operands to determine quadrent.
+- 0x129 (hypot3): Computes the hypotenuse of the three operands and stores the result in the first.
+
```

Comment:
My suggestion is to not read the C spec, but maybe to instead read the x87 ISA. Or any of the successors. Several of these instructions don't exist, either because they're redundant or because they're needlessly expensive to implement in hardware.